### ISSUE TITLE
After few hours of longevity run Commit Transaction is stuck without making progress

### ISSUE THREAD

**Issue by shiveshr** [pravega/pravega#3755]
**Problem description**
ref: #3749 

For smallTx longevity run, the commit transaction for a particular transaction gets stuck without making any progress. 
The client repeatedly checks for transaction's status and finds it to be in COMMITTING state. 

Upon investigating the logs, the commit reader has either not picked the event for processing or the previous processing has not completed fully. 

Commit transaction request has been accepted, and txn metadata is updated which means the event has already been posted into `commitStream`.
```
2019-05-02 18:13:25,373 12316936 [grpc-default-executor-103] INFO  i.p.c.s.r.g.v.ControllerServiceImpl - commitTransaction called for stream longevity/smallTx, txnId=lowBits: 8282
.
2019-05-02 18:13:25,382 12316945 [epollEventLoopGroup-2-3] INFO  i.p.controller.server.SegmentHelper - [requestId=370647] updateTableEntries request for _system/_tables/longevity/smallTx/transactionsInEpoch-0.#.23dbc143-e9b9-4a45-a1fc-3f93c9e6235a tableSegment completed.
```

There is a previous txn commit processing that is completing as this event is posted. 
At the end of txn commit workflow, we update txn metrics for which all open transactions are counted. 
Below logs indicates that the previous commit completed successfully. 

```
controller_2019-05-02.1.log:2019-05-02 18:13:25,397 12317965 [epollEventLoopGroup-2-1] INFO  i.p.controller.server.SegmentHelper - [requestId=138230] updateTableEntries request for _system/_tables/longevity/smallTx/metadata.#.23dbc143-e9b9-4a45-a1fc-3f93c9e6235a tableSegment completed.
controller_2019-05-02.1.log:2019-05-02 18:13:25,398 12317966 [epollEventLoopGroup-2-4] INFO  i.p.controller.server.SegmentHelper - [requestId=138231] readTableKeys _system/_tables/longevity/smallTx/epochsWithTransactions.#.23dbc143-e9b9-4a45-a1fc-3f93c9e6235a successful.
controller_2019-05-02.1.log:2019-05-02 18:13:25,399 12317967 [epollEventLoopGroup-2-3] INFO  i.p.controller.server.SegmentHelper - [requestId=138234] readTableKeys _system/_tables/longevity/smallTx/epochsWithTransactions.#.23dbc143-e9b9-4a45-a1fc-3f93c9e6235a successful.
controller_2019-05-02.1.log:2019-05-02 18:13:25,400 12317968 [epollEventLoopGroup-2-2] INFO  i.p.controller.server.SegmentHelper - [requestId=138233] readTableKeys _system/_tables/longevity/smallTx/transactionsInEpoch-0.#.23dbc143-e9b9-4a45-a1fc-3f93c9e6235a successful.
controller_2019-05-02.1.log:2019-05-02 18:13:25,402 12317970 [epollEventLoopGroup-2-3] INFO  i.p.controller.server.SegmentHelper - [requestId=138235] readTableKeys _system/_tables/longevity/smallTx/transactionsInEpoch-0.#.23dbc143-e9b9-4a45-a1fc-3f93c9e6235a successful.
```

We know the commit event was posted successfully because the call to commit transaction returned to the caller (test) successfully.
There are no logs indicating that the next commit event was ever picked up. 
Why the event was never picked up is not clear at the moment. 

**Problem location**
Commit Transaction workflow

**Suggestions for an improvement**
Find and fix the issue

**Comment by sumit-bm** [comment]
Ref: #3749

**Comment by sumit-bm** [comment]
Retested with build# 0.5.0-2269.6f8a820 and couldn't reproduce the issue

**Comment by shiveshr** [comment]
this is duplicate of a newer issue #3958 which was opened more recently. Labeling both of them for 0.6 as the severity of the issue is low and its easy to recover from it.

**Comment by RaulGracia** [comment]
Closed via #3985.
